### PR TITLE
Jenkins: add LPE CI test

### PR DESCRIPTION
**Pull request by TSC21**
Fixes https://github.com/PX4/Firmware/issues/9757. Also updates the ROS container tag.

### CODE REVIEW

**Comment by lamping7** [comment]
Why are we packaging LPE? Are we anticipating adding more missions to test against LPE? Packaging adds quite a bit of overhead and eats up a lot of space.

**Comment by TSC21** [comment]
> Why are we packaging LPE? Are we anticipating adding more missions to test against LPE? Packaging adds quite a bit of overhead and eats up a lot of space.

Actually yes. The idea would be add a VIO test were we could have LPE tested as well.

**Comment by lamping7** [comment]
So all we need is duplicate missions with different models I'm assuming.

**Comment by TSC21** [comment]
> The regular SITL build (posix_sitl_default) has both.

These have to do with the loaded configurations from `posix-configs`.

**Comment by TSC21** [comment]
Anyway, seems like current LPE doesn't pass on the tests on master @jgoppert. So probably something is not right on the pipeline.

**Comment by TSC21** [comment]
> So all we need is duplicate missions with different models I'm assuming.

Different models and different estimators.

**Comment by lamping7** [comment]
> These have to do with the loaded configurations from posix-configs.

Right. I understand that it is in the rcS. When I said model I mean vehicle, where vehicle is used to specify both the rcS and sdf. The way we launch ROS, we use the same vehicle arg, because we don't change the default estimator (which is ekf2).

See https://github.com/PX4/Firmware/blob/ddde968a6fa3e6d445bb5f3ceeb8f2ce775f076f/launch/posix_sitl.launch#L17

You're going to have to change the test files to include the `est` arg.

**Comment by TSC21** [comment]
> See https://github.com/PX4/Firmware/blob/ddde968a6fa3e6d445bb5f3ceeb8f2ce775f076f/launch/posix_sitl.launch#L17
>
>You're going to have to change the test files to include the `est` arg.

Oh ok. Makes more sense now!

File: Jenkinsfile [hunk 1]
```diff
@@ -562,10 +584,44 @@ pipeline {
...
+          }
+          options {
+            skipDefaultCheckout()
+          }
+          steps {
+            sh 'export'
+            sh 'rm -rf build; rm -rf px4-posix_sitl_lpe*; rm -rf .ros; rm -rf .gazebo'
+            unstash 'px4_sitl_lpe_package'
+            sh 'tar -xjpvf build/posix_sitl_lpe/px4-posix_sitl_lpe*.bz2'
+            sh 'px4-posix_sitl_lpe*/px4/test/rostest_px4_run.sh mavros_posix_test_mission.test mission:=multirotor_box vehicle:=iris'
```
Comment (lamping7) [inline Jenkinsfile]: est:=lpe

**Comment by lamping7** [comment]
OOPS. I lied. we have it in there.
https://github.com/PX4/Firmware/blob/ddde968a6fa3e6d445bb5f3ceeb8f2ce775f076f/test/mavros_posix_test_mission.test#L5

Just need to add the arg here:
https://github.com/PX4/Firmware/pull/9762/files#r197873115

**Comment by TSC21** [comment]
Well had to drop `process_logdata_ekf.py` step, since it's too specific for the EKF2 case - would be good to have something more generic for the different estimators, but for now, having the mission passing is enough for LPE case.

File: Jenkinsfile [hunk 2]
```diff
@@ -562,10 +562,43 @@ pipeline {
...
+            sh 'export'
+            sh 'rm -rf build; rm -rf px4-posix_sitl_default*; rm -rf .ros; rm -rf .gazebo'
+            unstash 'posix_sitl_package'
+            sh 'tar -xjpvf build/posix_sitl_default/px4-posix_sitl_default*.bz2'
+            sh 'px4-posix_sitl_default*/px4/test/rostest_px4_run.sh mavros_posix_test_mission.test mission:=multirotor_box vehicle:=iris est:=lpe'
+          }
+          post {
+            always {
+              sh 'px4-posix_sitl_default*/px4/Tools/upload_log.py -q --description "${JOB_NAME}: ${STAGE_NAME}" --feedback "${JOB_NAME} ${CHANGE_TITLE} ${CHANGE_URL}" --source CI .ros/rootfs/fs/microsd/log/*/*.ulg'
+              archiveArtifacts '.ros/**/*.pdf'
```
Comment (lamping7) [inline Jenkinsfile]: remove if not processing estimator data

File: Jenkinsfile [hunk 3]
```diff
@@ -562,10 +562,43 @@ pipeline {
...
+              archiveArtifacts '.ros/**/*.csv'
```
Comment (lamping7) [inline Jenkinsfile]: remove if not processing estimator data

Comment (TSC21) [reply Jenkinsfile]: @lamping7 these are outputs from the ekf log data processing or `upload_log.py`?

**Comment by TSC21** [comment]
@lamping7 done. @dagar CI passed. This is good to go as is. Though, we should plan on having a more generic log data processing.

File: Jenkinsfile [hunk 4]
```diff
@@ -167,7 +167,7 @@ pipeline {
               allowMissing: false,
               alwaysLinkToLastBuild: true,
               keepAll: true,
-              reportDir: 'build/scan-build/report_latest',
+              reportDir: 'build/scan-build/report_2018-06-19',
```
Comment (dagar) [inline Jenkinsfile]: ?

Comment (TSC21) [reply Jenkinsfile]: @dagar fixed

File: Jenkinsfile [hunk 5]
```diff
@@ -9,7 +9,7 @@ pipeline {
 
           def docker_base = "px4io/px4-dev-base:2018-03-30"
           def docker_nuttx = "px4io/px4-dev-nuttx:2018-03-30"
-          def docker_ros = "px4io/px4-dev-ros:2018-03-30"
+          def docker_ros = "px4io/px4-dev-ros:2018-06-19"
```
Comment (lamping7) [inline Jenkinsfile]: Why is this being changed as part of this PR? Aren't the changes to the container only needed for your odometry msg PR?

**Comment by TSC21** [comment]
> Is this all you want in this PR? If you fix the script to check the estimator logs, the containers will need changed again.

For now it is. I don't have much bandwidth to actually adapt the the estimator log analysis code for this, but should actually be a thing to consider in the future.

File: Jenkinsfile [hunk 6]
```diff
@@ -90,7 +90,7 @@ pipeline {
                     sh "make posix_sitl_default sitl_gazebo"
                     sh "make posix_sitl_default package"
                     sh "ccache -s"
-                    stash name: "px4_sitl_package", includes: "build/posix_sitl_default/*.bz2"
+                    stash name: "posix_sitl_package", includes: "build/posix_sitl_default/*.bz2"
```
[the remark this answers is not in the document]
Comment (TSC21) [reply Jenkinsfile]: yeah I guess we can keep it.

**Comment by lamping7** [comment]
I realize that's there, but I'm not sure it has much effect. Those values used for comparison seem arbitrary and I've never looked into what that analysis is doing and what the actual error values are.

**Comment by jgoppert** [comment]
@lamping7 I was the original author of all of that code. It is just checking if the average attitude error over the whole flight is less than 5 degrees currently.

**Comment by jgoppert** [comment]
@lamping7 @dagar  I had a look at the ekf2 analysis script to see what it is doing differently. The main thing that I noticed is that it isn't comparing to the ground truth data from sitl. I guess this is so that the script can be run on real flights as well, but the px4tools side can give you true error from the simulated state. This gives you some more fine grain feedback on estimator performance. The position error and standard deviation are also calculated in meters, so if you want to catch any performance drop in a PR you could start watching those numbers more closely.

**Comment by TSC21** [comment]
@dagar can we get this in?

We should probably discuss the CI approach for performance testing in a call or so.

**Pull request merged**